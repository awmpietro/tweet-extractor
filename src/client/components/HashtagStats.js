import React from 'react';
import { Table, CardTitle } from 'reactstrap';
import PropTypes from 'prop-types';

const HashtagStats = ({ stats }) => {
  const renderHashtags = () => {
    const hashtags = [];
    for (let i in stats.hashtags) {
      hashtags.push(
        <tr key={i}>
          <td>{i}</td>
          <td>{stats.hashtags[i]}</td>
        </tr>,
      );
    }
    return hashtags;
  };

  return (
    <>
      <CardTitle>
        <h4>Tweets by hashtags</h4>
      </CardTitle>
      <Table bordered striped>
        <thead>
          <tr>
            <th>Hashtag</th>
            <th>Quantity</th>
          </tr>
        </thead>
        <tbody>{renderHashtags()}</tbody>
      </Table>
    </>
  );
};

HashtagStats.propTypes = {
  stats: PropTypes.object,
};

export default HashtagStats;
