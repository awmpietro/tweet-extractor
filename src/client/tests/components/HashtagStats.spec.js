import React from 'react';
import HashtagStats from '../../components/HashtagStats';
import renderer from 'react-test-renderer';

describe('HashtagStats', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<HashtagStats stats={{}} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});