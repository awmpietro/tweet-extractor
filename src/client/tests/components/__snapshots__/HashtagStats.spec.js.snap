// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HashtagStats snapshot renders 1`] = `
Array [
  <div
    className="card-title"
  >
    <h4>
      Tweets by hashtags
    </h4>
  </div>,
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th>
          Hashtag
        </th>
        <th>
          Quantity
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
